radicle/src/git.rs

```diff
@@ -106,6 +113,48 @@ pub mod refs {
         pub fn id(remote: &RemoteId) -> Namespaced {
             IDENTITY_BRANCH.with_namespace(remote.into())
         }
+
+        /// The collaborative object reference, identified by `typename` and `object_id`, under the given `remote`.
+        ///
+        /// `refs/namespaces/<remote>/refs/cobs/<typename>/<object_id>`
+        ///
+        pub fn cob<'a>(
+            remote: &RemoteId,
+            typename: &cob::TypeName,
+            object_id: &cob::ObjectId,
+        ) -> Namespaced<'a> {
+            Qualified::from_components(
+                component!("cobs"),
+                Component::from(typename),
+                Some(object_id.into()),
+            )
+            .with_namespace(remote.into())
+        }
+
+        /// All collaborative objects, identified by `typename` and `object_id`, for all remotes.
+        ///
+        /// `refs/namespaces/*/refs/cobs/<typename>/<object_id>`
+        ///
+        pub fn cobs(typename: &cob::TypeName, object_id: &cob::ObjectId) -> PatternString {
+            refspec::pattern!("refs/namespaces/*")
+                .join(refname!("refs/cobs"))
+                .join(Component::from(typename))
+                .join(Component::from(object_id))
+        }
+
+        pub fn cob_suffix<R>(cob: &R) -> Option<(cob::TypeName, cob::ObjectId)>
```

Comment:
There probably isn't a better place for this right now, but I think we should eventually move it to a better home. Perhaps the type it returns, if common enough could have a `from_refstr` function that would be this function. Eg. `TypedObjectId::from_refstr(cob)`

Reply:
Mmmm ya, this could go in `radicle-cob` now that you mention it. I think generally a ref schema would always end in `<typename>/<object>`.

Reply:
Yeah, seems like a useful type to have. Going to merge this and let you implement that in a separate PR